services/business/checkers/route_rules/version_presence_checker_test.go

```diff
@@ -0,0 +1,124 @@
+package route_rules
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+)
+
+func TestCheckerWithPodsMatching(t *testing.T) {
+	assert := assert.New(t)
+	conf := config.NewConfig()
+	config.Set(conf)
+
+	// Setup mocks
+	podList := []v1.Pod{
+		fakePodsForLabels(map[string]string{"app": "reviews", "version": "v1", "stage": "production"}),
```

Comment:
Just to double check, a map[string]string it could be different than map[string]interface{}, so are you sure that k8s cast labels.Set as a map[string]string vs map[string]interface{} ?

If not in runtime it can skip some cast.

Reply:
labels.Set is a map[string]string by definition. So no problem in casting.
Dealing with IstioObjects is where most of the pain is. The definition of the struct is basically map[string]interface{}